### PR TITLE
Draft migration addressing issues in GH #41.

### PR DESCRIPTION
**Pull request by gsrohde**
This implements the changes outline in the spreadsheet at https://docs.google.com/spreadsheets/d/1fJgaOSR0egq5azYPCP0VRIWw1AazND0OCduyjONH9Wk/edit?pli=1#gid=1304822510 (see also GH #41 and related issues).

@dlebauer, @robkooper, @mdietze Please comment and suggest modifications if needed.

Marking `formats.mimetype_id` and `yields.entity_id` as foreign keys and imposing value constraints on the new data fields will probably be done in later migrations.

### CODE REVIEW

**Comment by gsrohde** [comment]
@dlebauer, @robkooper, @mdietze Please comment and suggest modifications if needed.

File: db/migrate/20150209174734_add_drop_tables_and_columns.rb [hunk 1]
```diff
@@ -0,0 +1,93 @@
+class AddDropTablesAndColumns < ActiveRecord::Migration
```
Comment (robkooper) [inline db/migrate/20150209174734_add_drop_tables_and_columns.rb]: Can it have a more specific name? Like add_constraint_mimetype?

**Comment by mdietze** [comment]
Looking at the Google Doc, on the Drop/Add page, what Issue is associated with dropping Start_date and End_date from the Runs table? Perhaps I'm being absent minded but I don't recall that discussion and can't remember why we'd want to drop those columns.

**Comment by dlebauer** [comment]
@mdietze https://ebi-forecast.igb.illinois.edu/redmine/issues/459 replaced *_date with *_time

**Comment by robkooper** [comment]
Who will update PEcAn to reflect the change to start_date and end_date in the code? Will there be a second pull request that goes with this issue?

**Comment by dlebauer** [comment]
> Who will update PEcAn to reflect the change to start_date and end_date in the code?

First we should resolve the question of whether there is a reason to make the change. I can't recall who suggested it or why this was suggested though it seems to have come up when Pat was here: in addition to  [Redmine 459](https://ebi-forecast.igb.illinois.edu/redmine/issues/459) it was also discussed in [Redmine 188](https://ebi-forecast.igb.illinois.edu/redmine/issues/188).

**Comment by mdietze** [comment]
Ok, that's bizarre, why would Pat have added start_time and end_time without migrating start_date and end_date? I had no idea that the database had BOTH and agree that one pair needs to be ditched. 

Looking at my copy of the database, it appears that start_time is  being used and start_date is completely empty. Therefore we should choose start_time over start_date, but I agree with @robkooper that we probably need a bit of time to verify that we're not using start_date or end_date anywhere in the code. Rob, this would be separate from PEcAn's internal use of variables named start_date and end_date since we use those to refer to datetime variables. Note that start_time is datetime and start_date is varchar so we're probably already OK.

File: db/migrate/20150209174734_add_drop_tables_and_columns.rb (hunk 1 above)
Comment (gsrohde) [inline db/migrate/20150209174734_add_drop_tables_and_columns.rb]: Sure, go for it.  I was having writer's block when I named that.

Comment (gsrohde) [inline db/migrate/20150209174734_add_drop_tables_and_columns.rb]: How about modify_formats_runs_yields?

**Comment by gsrohde** [comment]
@dlebauer I forget what our protocol is for marking a pull request as ready for me to review and pull.  Maybe "labelled with 'ready to review' AND assigned to me"?  And if so, how do I mark it when I'm not ready to accept a pull?  Assign it to some other person?

In any case, this pull request is still in the discussion stage while issues (not my issues, in this case) are being resolved, correct?

File: db/migrate/20150209174734_add_drop_tables_and_columns.rb (hunk 1 above)
Comment (robkooper) [inline db/migrate/20150209174734_add_drop_tables_and_columns.rb]: sounds good.

**Comment by gsrohde** [comment]
Rob: I renamed the migration (and did a few other tweaks).  Does this look OK now?

**Comment by robkooper** [comment]
@mdietze this will work for BU library with a single update:
```
delete from mimetypes where id=1000000003 or id=1000000001;
```

**Comment by robkooper** [comment]
+1

**Comment by mdietze** [comment]
@robkooper Just deleted these two records on the BU server, but am a bit confused where they came from in the first place since the BETY web interface doesn't have a mimetypes tab and in the db version on our server the formats table is referencing mime_type by character not id

**Comment by robkooper** [comment]
This migration will require a change to https://github.com/PecanProject/pecan/blob/master/db/R/dbfiles.R

**Comment by robkooper** [comment]
@mdietze Not sure where they came from, maybe left over from testing the dbfiles function one day.

**Comment by gsrohde** [comment]
Rob, is this ready to pull?  If so, assign it to me.

**Comment by dlebauer** [comment]
@robkooper what changes will need to be made to dbfiles.R? Is there an issue for this in the PEcAn repository?

**Comment by robkooper** [comment]
@dlebauer no there is no issue for this yet since I was unaware that columns are added/removed. Right now the code will add the mimetype to the dbfile, but will need to reference the values in mimetypes.

Easiest to see what happens is to install pecan (or the VM), do a git pull in bety and migrate the database. Next goto pecan and do a `scripts/build.sh --git --tests`. The tests should pass, and will hit a bunch of the database pieces.

**Comment by gsrohde** [comment]
@robkooper I reassigned this to you since your last note says "I would like to wait with the pull request until we have the appropriate code for PEcAn."

**Comment by dlebauer** [comment]
The appropriate changes for PEcAn are in pull request PecanProject/pecan#467

**Comment by gsrohde** [comment]
@robkooper (and @dlebauer): 

What is the status of this?  Shall I pull it?

**Comment by dlebauer** [comment]
from @robkooper 

> sure, needs to updated a bit to have all current code included.

**Pull request merged**